crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -369,6 +372,26 @@ where
         Ok(())
     }
 
+    /// Schedule a state change that restarts the active state.
+    /// This will fail if there is a scheduled operation
+    pub fn restart(&mut self) -> Result<(), StateError> {
+        if self.scheduled.is_some() {
+            return Err(StateError::StateAlreadyQueued);
+        }
+
+        let state = self.stack.last().unwrap();
+        self.scheduled = Some(ScheduledOperation::Set(state.clone()));
+        Ok(())
+    }
+
+    /// Same as [Self::restart], but if there is already a scheduled state operation,
+    /// it will be overwritten instead of failing
+    pub fn overwrite_restart(&mut self) -> Result<(), StateError> {
```

Comment:
Should this one still check if there is at least a state in the stack? Like `overwrite_pop`. Otherwise does it still needs to return a `Result`?

Reply:
No, there being at least one state is guaranteed.

Reply:
Then is the error `StateError::StackEmpty` ever relevant? There are a few things that could be removed from the code if not

Reply:
StackEmpty is only relevant for pop operations since you can't pop off the last state in the stack

Reply:
thanks for the explanation! so should `overwrite_restart` returns a `Result` when it doesn't have any error case?

Reply:
I kept it for consistency with the other methods, but it doesn't have to anymore.

Reply:
Removed the `Result` for now!